process/tech-due-diligence-guidelines.md

```diff
@@ -0,0 +1,103 @@
+  are.  If anything's unclear to you, reach out on the TOC mailing list for advice.
+* make sure you've read, in detail, the relevant [project proposal](https://github.com/quinton-hoole/toc/tree/master/proposals).
+  This will usually be in the form of an [open pull request](https://github.com/cncf/toc/pulls).
+  I'd suggest holding off on commenting on the PR until you've completed the next three steps.
+* take a look at some [previous submissions](https://github.com/cncf/toc/pulls?utf8=%E2%9C%93&q=is%3Apr)
+  (both successful and unsuccessful) to help calibrate your expectations.
+* do as much reading up as you need to (and consult with experts in the specific field) in order to familiarize yourself with the technology
+  landscape in the immediate vicinity of the project (and don't only use the proposal and that project's documentation as a guide in this regard).
+* at this point you should have a very clear technical idea of what exactly the project actually does and does not do, roughly how it compares with and differs from 
+  similar projects in it's technology area, and/or a set of unanswered questions in those regards. 
+* go through the [graduation criteria](https://www.cncf.io/projects/graduation-criteria/) and for each item,
+  decide for yourself whether or not you have enough info to make a strong, informed call on that item.
+  * If so, write it down, with motivation.
+  * If not, jot down what information you feel you're missing.
+  * Also take note of what unanswered questions the community might have posted in the PR review that you consider
+    to be critically important.
+    
+### Some example questions that will ideally need clear answers
+
+#### Technical
+
+* What are the current performance and scalability bounds of the software?  Have these been explicitly tested?
+* What exactly are the failure modes?  Are they well understood?  Have they been tested?  Do they form part of continuous integration testing?
+* What trade-offs have been made regarding performance, scalability, complexity, reliability, security etc?  Are these trade-offs explicit or implicit?  Why?
+* Code quality.  Does it look good, bad or mediocre to you (based on a spot review).  How thorough are the code reviews? Substance over form.
+  Are there explicit coding guidelines for the project?
+* What is the release model?  Versioning scheme?  Evidence of stability or otherwise of past stable released versions?
+* What is the CI/CD status?  Do explicit code coverage metrics exist? If not, what is the subjective adequacy of automated testing?  Do different levels of tests exist (e.g. unit, integration, interface, end-to-end), or is there only partial coverage in this regard?  Why? 
+
+#### Team
+
+* What is the general quality of informal communication around the project (slack, github issues, PR reviews, technical blog posts, etc)?
+* Would you hire the team? (this is of course a purely hypothetical question).  If not, why?
```

Comment:
Key high-level questions:

* Do we believe this is a growing, thriving project with committed contributors? 
* Is it aligned with CNCF's values and mission? 
* Do we believe it could eventually meet the graduation criteria? 
* Should it start at the inception level or incubation level?

Let's please tie this back to the proposal template and graduation criteria:

https://github.com/cncf/toc/blob/master/process/project_proposals.adoc
https://github.com/cncf/toc/blob/master/process/graduation_criteria.adoc

and with the TOC principles:
#47

Example more detailed questions:
* Does the project have a documented process for adding committers? 
* Does it have a documented governance model of any kind? 
* Does it have committers from multiple organizations?
* Does it have a code of conduct?
* Does it have a license? Which one? Does it have a CLA or DCO?

Let's focus on issues that will help us make a decision.

Reply:
excellent summary

Reply:
Agreed.  I added the above, and removed some of the more fluffy stuff.